src/examples/bad_code.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ros/ros.h"
+#include "std_msgs/Float64.h"
+
+double calc_pi    (void)
+{
+    ROS_INFO("calculating pi...");
+        double result;
+    for (int i=0; i < 100000000; ++i)
+    {
+        result += pow(-1, i)/(2*i+1);
+    }
+
+    ROS_INFO("done!");
+    return 4*result;
+}
+
+void convertCallback(const std_msgs::Float64::ConstPtr& msg)
+{
+    std_msgs::Float64 outmsg;
+
+        double radians = msg->data;
+    double degress = radians * (180/calc_pi());
+
+    outmsg.data = radians;
+
+    pub.publish(outmsg);
```

Comment:
pub is not not initialized in the convertCallback function. ros::Publisher pub must be moved outside the int main() function in order for this to compile properly